tests/listobjects/listobjects.go

```diff
@@ -129,143 +118,147 @@ func runTest(t *testing.T, test individualTest, params testParams, contextTupleT
-						Writes: &openfgav1.WriteRequestWrites{
-							TupleKeys: writeChunk,
-						},
-					})
-					require.NoError(t, err)
+		for stageNumber, stage := range test.Stages {
+			t.Run(fmt.Sprintf("stage_%d", stageNumber), func(t *testing.T) {
+				if contextTupleTest && len(stage.Tuples) > 20 {
+					// https://github.com/openfga/api/blob/05de9d8be3ee12fa4e796b92dbdd4bbbf87107f2/openfga/v1/openfga.proto#L151
+					t.Skipf("cannot send more than 20 contextual tuples in one request")
+				}
+				// arrange: write model
+				var typedefs []*openfgav1.TypeDefinition
+				model, err := parser.TransformDSLToProto(stage.Model)
+				if err != nil {
+					typedefs = oldparser.MustParse(stage.Model)
+				} else {
+					typedefs = model.GetTypeDefinitions()
 				}
-			}
 
-			if len(stage.ListObjectAssertions) == 0 {
-				t.Skipf("no list objects assertions defined")
-			}
+				writeModelResponse, err := client.WriteAuthorizationModel(ctx, &openfgav1.WriteAuthorizationModelRequest{
+					StoreId:         storeID,
+					SchemaVersion:   schemaVersion,
+					TypeDefinitions: typedefs,
+					Conditions:      model.GetConditions(),
+				})
+				require.NoError(t, err)
 
-			for _, assertion := range stage.ListObjectAssertions {
-				detailedInfo := fmt.Sprintf("ListObject request: %s. Model: %s. Tuples: %s. Contextual tuples: %s", assertion.Request, stage.Model, stage.Tuples, assertion.ContextualTuples)
+				tuples := stage.Tuples
+				tuplesLength := len(tuples)
+				// arrange: write tuples
+				if tuplesLength > 0 && !contextTupleTest {
+					for i := 0; i < tuplesLength; i += writeMaxChunkSize {
+						end := int(math.Min(float64(i+writeMaxChunkSize), float64(tuplesLength)))
+						writeChunk := (tuples)[i:end]
+						_, err = client.Write(ctx, &openfgav1.WriteRequest{
+							StoreId:              storeID,
+							AuthorizationModelId: writeModelResponse.GetAuthorizationModelId(),
+							Writes: &openfgav1.WriteRequestWrites{
+								TupleKeys: writeChunk,
+							},
+						})
+						require.NoError(t, err)
+					}
+				}
 
-				ctxTuples := assertion.ContextualTuples
-				if contextTupleTest {
-					ctxTuples = append(ctxTuples, stage.Tuples...)
+				if len(stage.ListObjectAssertions) == 0 {
+					t.Skipf("no list objects assertions defined")
 				}
 
-				// assert 1: on regular list objects endpoint
-				resp, err := client.ListObjects(ctx, &openfgav1.ListObjectsRequest{
-					StoreId:              storeID,
-					AuthorizationModelId: writeModelResponse.GetAuthorizationModelId(),
-					Type:                 assertion.Request.GetType(),
-					Relation:             assertion.Request.GetRelation(),
-					User:                 assertion.Request.GetUser(),
-					ContextualTuples: &openfgav1.ContextualTupleKeys{
-						TupleKeys: ctxTuples,
-					},
-					Context: assertion.Context,
-				})
+				for assertionNumber, assertion := range stage.ListObjectAssertions {
+					t.Run(fmt.Sprintf("assertion_%d", assertionNumber), func(t *testing.T) {
```

Comment:
Do you mean to ask if the first assertion on the YAML will be the first assertion to run? If so, in my experience, yes

Reply:
Yeah - if I'm looking at the YAML is the order the same as the unmarshaled output? I forget if we get preserved order.

Edit: I'm remembering now. Map order is not preserved, but slice order is (obviously). Since these assertions are array elements their order is preserved. I was being over paranoid and had a tired mind 😄